applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -318,6 +318,55 @@ void NavierStokesWallCondition<3,3>::GetDofList(DofsVectorType& rElementalDofLis
     }
 }
 
+template<unsigned int TDim, unsigned int TNumNodes>
+void NavierStokesWallCondition<TDim, TNumNodes>::Calculate(
+    const Variable< array_1d<double,3> >& rVariable,
+    array_1d<double,3>& rOutput,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    rOutput = ZeroVector(3);
+
+    if (rVariable == DRAG_FORCE) {
+        const auto& r_geom = GetGeometry();
+        const auto& r_integration_points = r_geom.IntegrationPoints(GeometryData::GI_GAUSS_2);
+        unsigned int n_gauss = r_integration_points.size();
+        Vector det_J_vect = ZeroVector(n_gauss);
+        r_geom.DeterminantOfJacobian(det_J_vect, GeometryData::GI_GAUSS_2);
+        const auto N_container = r_geom.ShapeFunctionsValues(GeometryData::GI_GAUSS_2);
+
+        // Calculate normal
+        array_1d<double,3> normal;
+        CalculateNormal(normal);
+        normal /= norm_2(normal);
+
+        // Finding parent element to retrieve viscous stresses
+        // Note that we assume in here that the shear stress is constant inside the element
+        auto& r_neighbours = this->GetValue(NEIGHBOUR_ELEMENTS);
+        KRATOS_ERROR_IF(r_neighbours.size() > 1) << "A condition was assigned more than one parent element." << std::endl;
+        KRATOS_ERROR_IF(r_neighbours.size() == 0) << "A condition was NOT assigned a parent element. "
+        << "This leads to errors for the slip condition [BEHR2004] "
+        << "Please execute the check_and_prepare_model_process_fluid process." << std::endl;
+
+        auto& r_parent = r_neighbours[0];
+        Vector shear_stress;
+        r_parent.Calculate(FLUID_STRESS, shear_stress, rCurrentProcessInfo);
+        array_1d<double,3> shear_stress_n;
+        ProjectViscousStress(shear_stress, normal, shear_stress_n);
+
+        // Loop the Gauss pts
+        for (unsigned int i_gauss = 0; i_gauss < n_gauss; ++i_gauss) {
+            const double w = det_J_vect[i_gauss] * r_integration_points[i_gauss].Weight();
+            double p = 0.0;
+            const auto& r_N = row(N_container, i_gauss);
+            for (unsigned int i_node = 0; i_node < r_geom.PointsNumber(); ++i_node) {
+                p += r_N[i_node] * r_geom[i_node].FastGetSolutionStepValue(PRESSURE);
+            }
+            rOutput += w * (p * normal - shear_stress_n);
```

Comment:
I'm a bit confused: Shouldn't the drag force include the shear stress tangential to the wall?

Reply:
It is including it... `shear_stress_n` is the normal projection of the viscous stress (the shear part of the Cauchy traction vector).

Reply:
Doesn't normal projection or multiplication of the shear stress with the normal vector of the wall mean, that there is only the normal contribution left and not the tangential one (2D), both tangential components (3D)?

Reply:
Also, I thought, that for drag calculation the normal part of the shear stress is not considered at all. I feel like I'm getting the code completely wrong.

Reply:
Nop, the drag is basically the integration of Cauchy traction vector over the surface, that is sigma · n. In other words, the projection of the stress over the surface of interest.

Concerning your last comment... if you not consider the shear stress what you measure is the pressure drag, but not the drag.

Reply:
I am aware of that and I am still wondering where the shear stress tangential to the surface is considered.

Reply:
But this is not the embedded contribution. This is the body-fitted condition. I think you are mixing both cases.